mobile/test/common/integration/client_integration_test.cc

```diff
@@ -227,10 +244,15 @@ TEST_P(ClientIntegrationTest, DISABLED_Http3WithQuicHints) {
   add_quic_hints_ = true;
 
   initialize();
+
+  auto address = fake_upstreams_[0]->localAddress();
+  default_request_headers_.setHost(absl::StrCat("localhost:", address->ip()->port()));
   default_request_headers_.setScheme("https");
-  basicTest();
-  // HTTP/3
-  ASSERT_EQ(3, last_stream_final_intel_.upstream_protocol);
+  // As we don't have certs for localhost the H3 handshake will fail.
+  basicTest(false);
```

Comment:
@danzh2010 suggested that if we use the default verifier then we can disable the validation check here:

https://source.corp.google.com/piper///depot/google3/third_party/envoy/src/api/envoy/extensions/transport_sockets/tls/v3/common.proto;l=524;rcl=572257956

Is that plausible?

Reply:
unfortunately not that simple.  even if we change the server to allow it, we end up with the mobile client rejecting the cert because of san mismatch.  when I comment that out we fail somewhere else :-(

Reply:
Blergh. Why is nothing simple?! Makes sense. Hopefully we'll be able to fix this in subsequent changes.